### PR TITLE
feat: added support for PrivateKeyAccount as signer in account config

### PR DESCRIPTION
**Pull request by VGabriel45**
# Summary

Added PrivateKeyAccount support as direct signer in account config.

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [x] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [x] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [ ] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [ ] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: packages/account/tests/account.spec.ts [hunk 1]
```diff
@@ -281,4 +280,33 @@ describe("Account Tests", () => {
...
+    const { bundlerUrl } = ganache;
+
+    const account = privateKeyToAccount(`0x${process.env.E2E_PRIVATE_KEY_ONE}`);
+
+    const createSmartAccount = createSmartAccountClient({
+      signer: account,
+      bundlerUrl,
+    });
+
+    await expect(createSmartAccount).rejects.toThrow("rpcUrl is required for PrivateKeyAccount signer type, please provide it in the config");
```
Comment (joepegler) [inline packages/account/tests/account.spec.ts:310]: Can you use the ERRORS config for this?

File: packages/common/src/utils/Helpers/convertSigner.ts [hunk 2]
```diff
@@ -51,7 +50,17 @@ export const convertSigner = async (signer: SupportedSigner, skipChainIdCalls: b
       }
       // convert viems walletClient to alchemy's SmartAccountSigner under the hood
       resolvedSmartAccountSigner = new WalletClientSigner(walletClient, "viem");
-      rpcUrl = walletClient?.transport?.url ?? null;
+      rpcUrl = walletClient?.transport?.url ?? undefined;
+    } else if (signer as PrivateKeyAccount) {
```
Comment (joepegler) [inline packages/common/src/utils/Helpers/convertSigner.ts:54]: I think this should be `instanceof` instead of `as`

**Review by joepegler — CHANGES_REQUESTED** [review]

File: packages/common/src/utils/Helpers/convertSigner.ts (hunk 2 above)
Comment (joepegler) [reply packages/common/src/utils/Helpers/convertSigner.ts:54]: You're casting every other signer to a PrivateKeyAccount here, - not just accounts.

File: packages/account/src/utils/Utils.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ export const compareChainIds = async (
   biconomySmartAccountConfig: BiconomySmartAccountV2Config,
   skipChainIdCalls: boolean,
 ): Promise<Error | void> => {
-  const signerResult = await convertSigner(signer, skipChainIdCalls);
+  const signerResult = await convertSigner(signer, skipChainIdCalls, biconomySmartAccountConfig.rpcUrl);
```
Comment (livingrockrises) [inline packages/account/src/utils/Utils.ts:58]: why is this added?

**Review by livingrockrises — COMMENTED** [review]
review i

File: packages/common/src/utils/Helpers/convertSigner.ts (hunk 2 above)
Comment (VGabriel45) [reply packages/common/src/utils/Helpers/convertSigner.ts:54]: instanceof works with classes, it does not work with types

Comment (VGabriel45) [reply packages/common/src/utils/Helpers/convertSigner.ts:54]: > You're casting every other signer to a PrivateKeyAccount here, - not just accounts.

Not sure what do you mean as this is the last "else if", it will cast it only if signer is indeed of PrivateKeyAccount type, otherwise it will break and throw an error.

File: packages/account/src/utils/Utils.ts (hunk 3 above)
Comment (VGabriel45) [reply packages/account/src/utils/Utils.ts:58]: If a user passes a PrivateKeyAccount there is no info about transport so he needs to at least pass rpcUrl in the config so we can properly construct the walletClient with the correct transport

**Review by livingrockrises — APPROVED** [review]

File: packages/common/src/utils/Helpers/convertSigner.ts (hunk 2 above)
Comment (VGabriel45) [reply packages/common/src/utils/Helpers/convertSigner.ts]: This was changed, please review

**Comment by joepegler** [comment]
LGTM. We can close this as I have added it here: https://github.com/bcnmy/biconomy-client-sdk/pull/463/commits/bb9ba8edfba18b667154309bc3098232b7bd79e9
